platform-db/src/test/java/com/softicar/platform/db/runtime/field/DbForeignFieldTest.java

```diff
@@ -118,43 +117,91 @@ public void testPrefetchingNullReferences() {
 	}
 
 	@Test
-	public void testPrefetchingLoadsOnlyStubObjectsAndInvalidatedObjects() {
+	public void testPrefetchingLoadsInvalidatedObjects() {
 
-		// insert objects initially
-		DbTestObject stubObject = insertObject("stub");
-		DbTestObject stubReferencingObject = insertObject(stubObject);
-		DbTestObject cachedObject = insertObject("cached");
-		DbTestObject cachedReferencingObject = insertObject(cachedObject);
-		DbTestObject invalidatedObject = insertObject("invalidated");
-		DbTestObject invalidatedReferencingObject = insertObject(invalidatedObject);
+		// setup: insert objects
+		DbTestObject object = insertObject("invalidated");
+		DbTestObject referencingObject = insertObject(object);
 
-		// clear caches
-		CurrentDbTableRowCacheMap.set(null);
+		// setup: invalidate the object
+		object.invalidate();
+
+		// setup: change the object in the database, circumventing the cache
+		new DbStatement()//
+			.addText("UPDATE " + DbTestObject.TABLE + " SET " + DbTestObject.STRING_FIELD + " = ?")
+			.addParameter("updated")
+			.executeUpdate();
+
+		// setup: validate preconditions
+		assertTrue(object.invalidated());
+		assertFalse(object.stub());
+		assertFalse(referencingObject.invalidated());
+		assertFalse(referencingObject.stub());
+
+		// execute: prefetch the referencing object
+		DbTestObject.FOREIGN_FIELD.prefetch(List.of(referencingObject));
+
+		// assert result
+		assertEquals("updated", referencingObject.getForeign().getString());
```

Comment:
actually, the call to `getString` de-invalidates the object in any case, so we don't test the prefeching in any way
```suggestion
		assertFalse(object.invalidated());
		assertEquals("updated", DbTestObject.STRING_FIELD.getValueDirectly(object));
```

Reply:
I see. Quite tough to see. Fixed.